### ISSUE TITLE
out of date dependencies in pulumi/pulumi container

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/pulumi-docker-containers#110]
The latest pulumi/pulumi container has some warnings about out of date tools. We should fix that. Do we have a process for keeping these tools up to date automatically? 

I see the following output when using npm in this container:
```json
[{"t":"2022-09-02T23:18:46.217571748Z","l":"npm notice New minor version of npm available! 8.15.0 -> 8.19.1\n"},{"t":"2022-09-02T23:18:46.217738791Z","l":"npm notice Changelog: <https://github.com/npm/cli/releases/tag/v8.19.1>\n"},{"t":"2022-09-02T23:18:46.217862849Z","l":"npm notice Run `npm install -g npm@8.19.1` to update!\n"},{"t":"2022-09-02T23:18:46.217948479Z","l":"npm notice \n"}]
```

**Comment by iwahbe** [comment]
We are not aware of a way to do this automatically. We should look into `snyk` logs for flagging this issue in the future.

**Comment by RobbieMcKinstry** [comment]
@iwahbe you can see the results of Snyk's scan [here](https://github.com/pulumi/pulumi-docker-containers/runs/8197523120?check_suite_focus=true#step:5:18).

**Comment by RobbieMcKinstry** [comment]
Hello hello!

I looked into this at the beginning of the quarter and never followed up. I installed Snyk into our container registry and inspected the warnings. There's good news and bad news.

The good news is that for dependencies like npm, we fetch the latest version in `apt` each time we rebuild the container (i.e. each time we cut a new release). This is true not just for NPM but also for almost every dependency installed. By now, I expect the warning you reported to have gone away. Snyk indicated that a new version was pushed to apt. 

The bad news is that we _can't_ address these issues automatically or even take any action against them because of our versioning policy. Because container images are expected to be immutable, we can't refresh the dependencies on a container and re-push the image for a given tag. The container will be updated to have all of the latest installations by the time of the next release. But in between each release, no code change to the provision script will update version of the out of date dep, and even if it did, we wouldn't have a way to release it!

We should look into decoupling the container image versions from the CLI version. If we want `@pulumi/pulumi:v3.41.0` to track the CLI at v3.41.0, we can keep that release line as is. But we should offer a `nightly` release line or something similar to get the latest security updates as soon as they drop. I'm going to open a separate issue for that.

I'm going to close this issue because (1) its not actionable (2) the original item should have been addressed.
